File: exercism-csharp/resistor-color/ResistorColor.cs
public static class ResistorColor
{
    private static readonly string[] Cols;

    static ResistorColor() =>
        Cols = [
            "black",
            "brown",
            "red",
            "orange",
            "yellow",
            "green",
            "blue",
            "violet",
            "grey",
            "white"
        ];

    public static int ColorCode(string color)
    {
        // return Array.IndexOf(Cols, color);
        for (int i = 0; i < Cols.Length; i++)
            if (Cols[i] == color)
                return i;

        return -1;
    }

    public static string[] Colors() => Cols;
}

File: exercism-cpp/resistor-color/resistor_color.cpp
#include "resistor_color.h"
#include <algorithm>

namespace resistor_color {
    std::vector<std::string> ori_colors = {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    };

    long long color_code(const std::string &color) {
        return std::distance(ori_colors.cbegin(), std::find(ori_colors.cbegin(), ori_colors.cend(), color));
    }

    const std::vector<std::string> & colors() {
        return ori_colors;
    }
} // namespace resistor_color
